src/main/java/tes/taskmanager/Storage.java

```diff
@@ -0,0 +1,107 @@
+    private static final DateTimeFormatter FORMATTER_RECEIVE = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    /** Format of the date and time in database */
+    private static final DateTimeFormatter FORMATTER_STORE = DateTimeFormatter.ofPattern("MMM dd yyyy hh:mm a");
+
+    /**
+     * Constructs an empty list to load the tasks stored in local disk.
+     *
+     */
+    public Storage() {
+        this.storeList = new ArrayList<>();
+    }
+
+    /**
+     * Saves the recorded tasks into the local disk.
+     * If there is no existing directory mentioned in the path, create one.
+     */
+    public static void saveToFile(ArrayList<Task> tasks) {
+        try {
+            File file = new File(FILE_PATH);
+            file.getParentFile().mkdirs();
+            PrintWriter writer = new PrintWriter(file);
+
+            for (Task task : tasks) {
+                writer.println(task.toString());
+            }
+            writer.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("Error writing to the file: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Loads existing task list from the local disk if there is one.
+     * If no, a new empty task list is used.
+     */
+    public ArrayList<Task> loadFromFile() {
+        try {
+            File file = new File(FILE_PATH);
+            if (!file.exists()) {
+                return new ArrayList<>();
+            }
+            Scanner scanner = new Scanner(file);
+
+            while (scanner.hasNext()) {
+                String line = scanner.nextLine();
+                try {
+                    if (line.contains("from:")) {
+                        String[] split = line.split(" \\| ");
+                        String temp = split[3].substring(6);
+                        String[] time = temp.split(" to: ");
+                        String from = time[0];
+                        String to = time[1];
+                        LocalDateTime from1 = LocalDateTime.parse(from, FORMATTER_STORE);
+                        LocalDateTime to1 = LocalDateTime.parse(to, FORMATTER_STORE);
+                        Event store = new Event(split[2], from1, to1);
+                        if (split[1].equals("X")) {
+                            store.mark();
+                        }
+                        storeList.add(store);
+                    } else if (line.contains("by:")) {
+                        String[] split = line.split(" \\| ");
+                        String by = split[3].substring(4);
+                        LocalDateTime by1 = LocalDateTime.parse(by, FORMATTER_STORE);
+                        Deadline store = new Deadline(split[2], by1);
+                        if (split[1].equals("X")) {
+                            store.mark();
+                        }
+                        storeList.add(store);
+                    } else {
+                        String[] split = line.split(" \\| ");
```

Comment:
I think a better naming convention would be to replace `split` with a plural form of a noun, perhaps something like `segments`.